src/geo/data-providers/geojson/geojson-data-provider.js

```diff
@@ -15,33 +15,37 @@ var GeoJSONDataProvider = function (vectorLayerView, layerIndex) {
 // TODO: We can extract each "generator" to an individual file so that this file doesn't get too BIG
 GeoJSONDataProvider.prototype._dataGeneratorsForDataviews = {
   category: function (features, options) {
+    var filter = this._vectorLayerView.renderers[this._layerIndex].filter;
```

Comment:
Done in https://github.com/CartoDB/d3.cartodb/commit/e707b3d22a732a2952dfa798dd14531ba8b19966